_posts/2020-05-19-pathfinding.md

```diff
@@ -0,0 +1,149 @@
+
+Il problema in esame si chiama "ricerca del cammino", o più comunemente "pathfinding" o "path planning" per gli amici controllisti. 
+È nato nel contesto della robotica, dalla quale derivano molte delle assunzioni alla base, e dei controlli automatici, con i quali ha molti più rapporti di quanto piacerebbe a ogni informatico che si rispetti il quale, per legge, deve odiare i controllisti e tutto ciò che li riguarda. 
+
+Nei successivi articoli di questa serie ci saranno un po' di controlli automatici, ma mi ritengo un informatico che si rispetti, quindi ne parlerò molto male, puoi stare tranquillo. 
+
+
+## Rappresentazione delle mappe
+
+In questo articolo ci saranno pochi controlli automatici, grazie al cielo. 
+Però ci saranno i grafi, e quelli piacciono a persone orribili, come i controllisti, i matematici e gli statistici.
+Per qualche motivo, però, anche gli informatici sono riusciti a farseli piacere un sacco, forse perché sono così dannatamente utili. 
+Ma cos'è un grafo? 
+
+### Elementi di teoria dei grafi
+
+Non me ne vogliano i matematici per quelllo che sto per dire: un *grafo* è un insieme di *nodi* associati tra di loro in una maniera particolare. 
+Un'associazione tra due nodi si chiama *arco* e può avere o meno una *direzione* e un *peso*.
+Due nodi associati si dicono *vicini*.
+
+"Eh?"
+
+Ci riprovo.
+Questo è un grafo (immagine da Wikipedia):
+
+![graph]({{site.baseurl}}/assets/img/graph.png){:width="30%"}
+
+In particolare, è un grafo *non diretto* perché gli archi non hanno una direzione, quindi se A è vicino di B allora anche B è vicino di A.
+Se fosse *diretto* gli archi avrebbero delle freccine a uno dei propri terminali e l'associazione tra archi non sarebbe (necessariamente) simmetrica. 
+
+Questo è anche un grafo *non pesato* perché un nodo può solo essere o non essere vicino di un altro. 
+Se fosse *pesato* gli archi avrebbero dei numeri in mezzo e A potrebbe essere vicino a B con peso 15, mentre C potrebbe essere vicino a D con peso 18.
+La cosa interessante è che nei grafi diretti pesati, A può essere vicino a B con peso 16 e B può essere vicino ad A con peso 29.
+
+Non preoccuparti, comunque: nei videogiochi, nella maggior parte dei casi, i grafi sono tutti non diretti, pesati, con due soli valori dei pesi: 1 e $$\sqrt{2}$$. 
+
+Perché la radice?
+Eh, arrivo. 
+Tienila a mente, ma sopporta un altro po' di definizioni.
+
+Un insieme di archi che parte da un nodo A e arriva a un nodo B si chiama *cammino*. 
+Un cammino ha una certa *lunghezza* o *costo*, che nel caso dei grafi non pesati è il numero di archi che compongono il cammino mentre nei grafi pesati è la somma dei pesi degli archi che compongono il cammino.
+Ad esempio, nel grafo mostrato in figura esiste un cammino di lunghezza 3 tra i nodi 1 e 6: 1->5->4->6. 
+Ovviamente puoi arrivare da 1 a 6 anche passando per i nodi 1->2->3->4->6. 
```

Comment:
I would probably wraps examples like those on code on a single line like `1->2` kind of thing